### PR TITLE
Warn if duplicate key is found in record

### PR DESCRIPTION
**Pull request by wojtekmach**
Similar to #9452 :D

Worth noting erlc errors on this too:

```
-record(r, {a, a}).
```

src/m.erl:2: field a already defined in record r

### CODE REVIEW

File: lib/elixir/test/elixir/record_test.exs [hunk 1]
```diff
@@ -262,6 +263,15 @@ defmodule RecordTest do
     end
   end
 
+  test "duplicate keys" do
+    assert capture_io(:stderr, fn ->
```
Comment (fertapric) [inline lib/elixir/test/elixir/record_test.exs]: Capturing a named device other than `:stdio`, such as `:stderr`, happens globally and requires `async: false`.

In this case, I would move the test to the `warning_test.exs` file.

Comment (wojtekmach) [reply lib/elixir/test/elixir/record_test.exs]: Ah, right. Fixed!

File: lib/elixir/test/elixir/kernel/warning_test.exs [hunk 2]
```diff
@@ -1774,14 +1774,27 @@ defmodule Kernel.WarningTest do
...
-               defstruct [:foo, :foo, :bar]
+               defstruct [:foo, :bar, foo: 1]
              end
              """)
            end) =~ "duplicate key :foo found in struct"
   after
     purge(TestMod)
   end
 
+  test "duplicate keys" do
```
Comment (fertapric) [inline lib/elixir/test/elixir/kernel/warning_test.exs]: Nitpick:

```suggestion
  test "defrecord warns with duplicate keys" do
```

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**